microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,156 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
 
     return [u for u in uncovered if (u['sequence'] != '') &
                                     (len(u['sequence']) >= min_subseq_len)]
+
+
+def get_q_id(hit):
+    """ Returns the query ID for a hit.
+
+    Parameters
+    ----------
+    A hit parsed from an HHsearch output file, i.e. dict with at least the key
+    'alignment' which is a dict by itself and comes at least with the key
+    'Q xxx' where xxx is some identifier. The value for this 'Q xxx' key is a
+    third dict which needs to contain the key 'sequence'.
+
+    Returns
+    -------
+    str : The query ID starting with 'Q '.
+    """
+    # find the right ID
+    id_ = [id_ for id_ in hit['alignment'].keys()
```

Comment:
The variable names look bit awkward. Will there only be one valid id of all `hit['alignment'].keys()`? If no, then this code cannot guarantee that the ids are in order. If yes, this is why you have `[0]` in the end. Right? Looks bit awkward again. How about for loop and break at the first match?